src/ovm/ObolValidatorManager.sol

```diff
@@ -280,16 +283,16 @@ contract ObolValidatorManager is OwnableRoles {
     if (remainingFee > 0) payable(msg.sender).transfer(remainingFee);
   }
 
-  /// Recover non-OWR tokens to a recipient
-  /// @param nonOWRToken Token to recover (cannot be OWR token)
+  /// Recover non-OVM tokens to a recipient
+  /// @param nonOVMToken Token to recover (cannot be OVM token)
   /// @param recipient Address to receive recovered token
-  function recoverFunds(address nonOWRToken, address recipient) external payable {
+  function recoverFunds(address nonOVMToken, address recipient) external onlyOwnerOrRoles(RECOVER_TOKEN_ROLE) {
```

Comment:
I'm not sure we should have both a role and an option to set a recovery address in the constructor, it feels a bit confused imo. TBH, the idea of there being two options on the default recovery (i.e. picking if principal or reward should get it) is probably a bad idea in retrospect. 

I accept that a permissionless function to an arbitrary address could someday be unsafe. We need to keep the arbitrary addresses (i.e. the tokens that get airdropped), so our option is almost definitely to restrict the caller. 

Maybe we get rid of the settable `recoveryAddress` in the constructor, we keep the new role, and allow arbitrary destinations? so `onlyOwnerOrRole(RECOVER_TOKEN_ROLE)` can send any token to any address? and we scrap the logic about choosing principal or recipient?

Reply:
> Maybe we get rid of the settable recoveryAddress in the constructor, we keep the new role, and allow arbitrary destinations?

I support this

Reply:
Done as discussed.